backend/src/main/java/org/eclipse/tractusx/semantics/registry/controller/AssetAdministrationShellApiDelegate.java

```diff
@@ -86,21 +87,18 @@ public ResponseEntity<Void> deleteSubmodelDescriptorByIdThroughSuperpath( String
     @Override
     public ResponseEntity<GetAssetAdministrationShellDescriptorsResult> getAllAssetAdministrationShellDescriptors( Integer limit, String cursor,
           AssetKind assetKind, String assetType ) {
-        Integer page = 0 ;
-        Integer pageSize = 100;
-        ShellCollectionDto dto =  shellService.findAllShells(page, pageSize);
+        ShellCollectionDto dto =  shellService.findAllShells(limit, cursor);
         GetAssetAdministrationShellDescriptorsResult result = shellMapper.toApiDto(dto);
+        result.setPagingMetadata( new PagedResultPagingMetadata().cursor( dto.getCursor() ) );
```

Comment:
It would make sense to do this set in the service class and not in the controller. All relevant information is available in the service layer.